src/zope/configuration/xmlconfig.py

```diff
@@ -80,41 +79,40 @@ class ZopeXMLConfigurationError(ConfigurationError):
     Example
 
         >>> from zope.configuration.xmlconfig import ZopeXMLConfigurationError
-        >>> v = ZopeXMLConfigurationError("blah", AttributeError, "xxx")
+        >>> v = ZopeXMLConfigurationError("blah", AttributeError("xxx"))
         >>> print(v)
         'blah'
-            AttributeError: xxx
+    	    AttributeError: xxx
     """
-    def __init__(self, info, etype, evalue):
-        self.info, self.etype, self.evalue = info, etype, evalue
 
-    def __str__(self):
-        # Only use the repr of the info. This is because we expect to
-        # get a parse info and we only want the location information.
-        return "%s\n    %s: %s" % (
-            repr(self.info), self.etype.__name__, self.evalue)
+    USE_INFO_REPR = True
+
 
-class ZopeSAXParseException(ConfigurationError):
+class ZopeSAXParseException(ConfigurationWrapperError):
     """
     Sax Parser errors, reformatted in an emacs friendly way.
 
     Example
 
         >>> from zope.configuration.xmlconfig import ZopeSAXParseException
-        >>> v = ZopeSAXParseException("foo.xml:12:3:Not well formed")
+        >>> v = ZopeSAXParseException(Exception("foo.xml:12:3:Not well formed"))
         >>> print(v)
         File "foo.xml", line 12.3, Not well formed
+    	    Exception: foo.xml:12:3:Not well formed
     """
-    def __init__(self, v):
-        self._v = v
 
-    def __str__(self):
-        v = self._v
-        s = tuple(str(v).split(':'))
+    def __init__(self, exception):
+        s = tuple(str(exception).split(':'))
         if len(s) == 4:
-            return 'File "%s", line %s.%s, %s' % s
+            info = 'File "%s", line %s.%s, %s' % s
         else:
-            return str(v)
+            info = str(exception)
+        super(ZopeSAXParseException, self).__init__(info, exception)
```

Comment:
TBH, I don't actually see the point of this exception anymore. My emacs is able to follow links to 'foo.xml:12:3' just fine, and it's not like we consistently reformat the parser location information. 

Maybe we can just drop this reformatting?

Reply:
In fact, when I load malformed ZCML in a compile buffer, only the *original* form of the error is highlighted by emacs as something it can jump to. The reformat we're doing here is not.

Reply:
I suspect whoever wrote this code had their emacs configured to parse Python traceback lines.

I'm +1 for dropping the reformatting.